### PR TITLE
Implement host metrics

### PR DESCRIPTION
**Pull request by ThouCheese**
Implements host metrics. I went for the same API structure that I used for node metrics, even though we are usually going to send the metrics for one host at a time.

### CODE REVIEW

File: blockjoy/api/v1/metrics.proto [hunk 1]
```diff
@@ -24,15 +25,36 @@ message Metrics {
 
 message NodeMetricsResponse {}
```
Comment (tstaetter) [inline blockjoy/api/v1/metrics.proto:26]: empty response?

File: blockjoy/api/v1/metrics.proto [hunk 2]
```diff
@@ -24,15 +25,36 @@ message Metrics {
...
 
+// This message is used to store the metrics for a given set of hosts.
 message HostMetricsRequest {
-    // todo
+    // This field maps the id of the host to the metrics that apply to that host.
+    map<string, HostMetrics> metrics = 1;
```
Comment (tstaetter) [inline blockjoy/api/v1/metrics.proto:31]: what are u actually requesting here?

Comment (ThouCheese) [reply blockjoy/api/v1/metrics.proto:31]: A map of host ids to the metrics for those hosts. It is the same as the request for node where there is a map from node_id to node metrics.

File: blockjoy/api/v1/metrics.proto [hunk 3]
```diff
@@ -24,15 +25,36 @@ message Metrics {
...
+}
+
+// The metrics for a single `Host`. Note that there is no host id in this message, because it is
+// embedded in the `HostMetricsRequest`, where the key of the map already is the id of the host.
+message HostMetrics {
```
[the remark this answers is not in the document]
Comment (ThouCheese) [reply blockjoy/api/v1/metrics.proto:36]: This is not returned, it is part of the request

File: blockjoy/api/v1/metrics.proto (hunk 1 above)
Comment (tstaetter) [reply blockjoy/api/v1/metrics.proto:26]: if it's really empty, use `google.protobuf.Empty` as return type

File: blockjoy/api/v1/metrics.proto [hunk 4]
```diff
@@ -24,15 +25,36 @@ message Metrics {
...
+    // core that is used. Example: if there are 4 cores, and each is at 78% usage, the number sent
+    // will be 312.
+    optional uint16 cpu = 1;
+    // This is the amount of memory used, given in bytes.
+    optional uint64 mem = 2;
+    // This is the amount of disk used, given in bytes.
+    optional uint64 disk = 3;
+    // This is the load, given as a percentage. Example: if the load given by top is
+    // `2.73, 2.96, 2.53`, then the value sent here will be 2.96.
+    optional double load = 4;
```
Comment (loony-bean) [inline blockjoy/api/v1/metrics.proto:47]: we need 3 load values, it allows to get important insights about the load
load_avg_1m, load_avg5m, etc

File: blockjoy/api/v1/metrics.proto [hunk 5]
```diff
@@ -24,15 +25,36 @@ message Metrics {
...
+    // This is the number of bytes (not bits) sent over the network since the start of the node.
+    // This means that this value is going to increase monotonically when the node starts.
+    optional uint64 network = 5;
```
Comment (loony-bean) [inline blockjoy/api/v1/metrics.proto:50]: we need both tx and rx values

File: blockjoy/api/v1/metrics.proto (hunk 1 above)
Comment (ThouCheese) [reply blockjoy/api/v1/metrics.proto:26]: I didn't even touch this part :cry:

File: blockjoy/api/v1/metrics.proto (hunk 2 above)
Comment (tstaetter) [reply blockjoy/api/v1/metrics.proto:31]: 👍🏻

File: blockjoy/api/v1/metrics.proto [hunk 6]
```diff
@@ -3,36 +3,66 @@ syntax = "proto3";
...
+    // The amount of CPU that is currently in use. This is given as the sum of percentages of each
+    // core that is used. Example: if there are 4 cores, and each is at 78% usage, the number sent
+    // will be 312.
+    optional uint32 used_cpu = 1;
+    // This is the amount of memory used, given in bytes.
+    optional uint64 used_memory = 2;
+    // This is the amount of disk used, given in bytes.
+    optional uint64 used_disk_space = 3;
+    // This is the load during the last minute, given as a percentage. Example: if the load given
+    // by top is `2.73, 2.96, 2.53`, then the value sent here will be 2.73.
```
Comment (loony-bean) [inline blockjoy/api/v1/metrics.proto:45]: c/p docs

Comment (loony-bean) [reply blockjoy/api/v1/metrics.proto:45]: oh sorry i misread

Comment (loony-bean) [reply blockjoy/api/v1/metrics.proto:45]: .. still this part is c/p
> then the value sent here will be 2.73.

**Pull request merged**